### PR TITLE
feat:improve ui of Github

### PR DESCRIPTION
**Pull request by rkyrahulkumar**

### CODE REVIEW

File: github_access/templates/github_access/access_request_form.html [hunk 1]
```diff
@@ -1,110 +1,256 @@
...
-      if(!labelComponents.length){
-        labelComponents = [{}]
-      }
-      labelComponents[0]["repoList"] = getSelectedRepos()
-      $('#github_accessLabel').val(JSON.stringify(labelComponents));
+  };
+
+  const selectGithubRepoCheckbox = (elem, checked) => {
+    if(checked) {
+      $(elem).find('div').show();
```
Comment (kneeraj2000) [inline github_access/templates/github_access/access_request_form.html:28]: remove these and corresponding div

File: github_access/templates/github_access/access_request_form.html [hunk 2]
```diff
@@ -1,110 +1,256 @@
...
+    for(iter = 0; iter < repos.length; iter++) {
+      removeRepoSelectionSpanElem(repos[iter], $("#github-repo-table").find(`tr[repo_id="${$(repos[iter]).attr('repo_id')}"]`));
+    }
+    $("#selectAllGithubRepo").prop("checked", false);
+    updateSelectedGithubRepo();
+  };
+
+  const selectAllGithubRepo = () => {
+    const repos = $('#github-repo-table').children('tr');
+
```
Comment (kneeraj2000) [inline github_access/templates/github_access/access_request_form.html:83]: need to call updateSelectedGithubRepo() here as well

File: github_access/templates/github_access/access_request_form.html [hunk 3]
```diff
@@ -1,110 +1,256 @@
...
+  }
+
+  const removeGithubReposelection = (elem) => {
+    $("#selectAllGithubRepo").prop("checked", false);
+    removeRepoSelectionSpanElem($("#github-repo-selection-table").find(`tr[repo_id="${$(elem).attr('repo_id')}"]`), elem);
+  };
+
+  const removeGithubReposelectionUI = (elem) => {
+    rightElem = elem.parentElement.parentElement;
+    removeRepoSelectionSpanElem(rightElem, $("#github-repo-table").find(`tr[repo_id="${$(rightElem).attr('repo_id')}"]`));
```
Comment (kneeraj2000) [inline github_access/templates/github_access/access_request_form.html:107]: need to change selectAll toggle to false and call updateSelectedGithubRepo()

File: github_access/templates/github_access/access_request_form.html [hunk 4]
```diff
@@ -1,110 +1,256 @@
...
-          <option value="{{repo}}">{{repo}}</option>
-          {%endfor%}
-        </select>
+<h1 class="pt-5 text-sm text-gray-900">Search Git repository</h1>
+<div class="grid grid-cols-2 gap-4">
+  <div class="flex-col">
+    <div class="mt-4 flex flex-col">
+      <div class="flex felx-row">
+        <div class="flex-auto relative flex items-center">
+          <input type="text" name="search" id="search" class="block w-full rounded-md border-gray-300 pr-12 shadow-sm focus:border-blue-600 focus:ring-blue-600 sm:text-sm" placeholder="Search github repos">
```
Comment (kneeraj2000) [inline github_access/templates/github_access/access_request_form.html:160]: please change id to gitSearch

File: github_access/templates/github_access/access_request_form.html [hunk 5]
```diff
@@ -1,110 +1,256 @@
...
-            class="fa fa-angle-double-right" style="font-size:24px"></i></button>
-        <button type="button" id="search-github_rightSelected" class="btn btn-block btn-primary"><i
-            class="fa fa-angle-right" style="font-size:24px"></i></button>
-        <button type="button" id="search-github_leftSelected" class="btn btn-block btn-primary"><i
-            class="fa fa-angle-left" style="font-size:24px"></i></button>
-        <button type="button" id="search-github_leftAll" class="btn btn-block btn-primary" disabled><i
-            class="fa fa-angle-double-left" style="font-size:24px"></i></button>
+                           <input type="checkbox" class="absolute left-4 top-1/2 -mt-2 h-4 w-4 rounded border-gray-300 text-blue-600 focus:ring-blue-600 sm:left-6" onclick="handleGithubRepoSelectionCheckbox(this)"></input>
+                         </td>
+                         <td class="whitespace-nowrap py-4 pr-3 text-sm font-medium text-gray-900" id="description-td">{{repo.orgName}}/{{repo.repoName}}</td>
```
Comment (kneeraj2000) [inline github_access/templates/github_access/access_request_form.html:187]: please append git at the starting of the ids

File: github_access/templates/github_access/access_request_form.html (hunk 3 above)
Comment (sharad-sharma) [reply github_access/templates/github_access/access_request_form.html:107]: Done.

File: github_access/templates/github_access/access_request_form.html (hunk 2 above)
Comment (sharad-sharma) [reply github_access/templates/github_access/access_request_form.html:81]: Done.

**Review by kneeraj2000 — DISMISSED** [review]

File: github_access/access.py [hunk 6]
```diff
@@ -229,18 +234,18 @@ def fetch_access_approve_email(self, request, data):
     def validate_request(self, access_labels_data, request_user, is_group=False):
```
Comment (niveshm) [inline github_access/access.py:234]:
```suggestion
    def validate_request(self, access_request_form, request_user, is_group=False):
```

**Review by niveshm — APPROVED** [review]

**Pull request merged**